ComposureWatch/src/Components/RankIcon.jsx

```diff
@@ -38,11 +39,7 @@ const RankIcon = ({ rank }) => {
     },
   ];
 
-  let charRank;
-  let result = rankList.find((item) => item.id === rank);
-  if (result) {
-    charRank = result.img;
-  }
+  let charRank = SearchImg(rank, rankList);
```

Comment:
Do you want to do any special handling the rank passed is not found? Maybe just a placeholder image or something?